demos/interactive_face_detection_demo_gapi/visualizer.cpp

```diff
@@ -0,0 +1,355 @@
+    return size;
+}
+
+void EmotionBarVisualizer::draw(cv::Mat& img, std::map<std::string, float> emotions, cv::Point org, cv::Scalar fgcolor, cv::Scalar bgcolor) {
+    cv::Mat tmp = img(cv::Rect(org.x, org.y, size.width, size.height));
+    cv::addWeighted(tmp, 1.f - opacity, bgcolor, opacity, 0, tmp);
+
+    auto drawEmotion = [&](int n, std::string text, float value) {
+        cv::Point torg(org.x + padding.width, org.y + n * ystep + textSize.height + padding.height);
+
+        int textWidth = textSize.width + 10;
+        cv::Rect r(torg.x + textWidth, torg.y - textSize.height, size.width - 2 * padding.width - textWidth, textSize.height + textBaseline / 2);
+
+        cv::putText(img, text, torg, cv::FONT_HERSHEY_COMPLEX_SMALL, textScale, fgcolor, textThickness);
+        cv::rectangle(img, r, fgcolor, 1);
+        r.width = static_cast<int>(r.width * value);
+        cv::rectangle(img, r, fgcolor, cv::FILLED);
+    };
+
+    for (size_t i = 0; i< emotionNames.size(); i++) {
+        drawEmotion(i, emotionNames[i], emotions[emotionNames[i]]);
+    }
+}
+
+// PhotoFrameVisualizer
+PhotoFrameVisualizer::PhotoFrameVisualizer(int bbThickness, int photoFrameThickness, float photoFrameLength):
+    bbThickness(bbThickness), photoFrameThickness(photoFrameThickness), photoFrameLength(photoFrameLength) {
+}
+
+void PhotoFrameVisualizer::draw(cv::Mat& img, cv::Rect& bb, cv::Scalar color) {
+    cv::rectangle(img, bb, color, bbThickness);
+
+    auto drawPhotoFrameCorner = [&](cv::Point p, int dx, int dy) {
+        cv::line(img, p, cv::Point(p.x, p.y + dy), color, photoFrameThickness);
+        cv::line(img, p, cv::Point(p.x + dx, p.y), color, photoFrameThickness);
+    };
+
+    int dx = static_cast<int>(photoFrameLength * bb.width);
+    int dy = static_cast<int>(photoFrameLength * bb.height);
+
+    drawPhotoFrameCorner(bb.tl(), dx, dy);
+    drawPhotoFrameCorner(cv::Point(bb.x + bb.width - 1, bb.y), -dx, dy);
+    drawPhotoFrameCorner(cv::Point(bb.x, bb.y + bb.height - 1), dx, -dy);
+    drawPhotoFrameCorner(cv::Point(bb.x + bb.width - 1, bb.y + bb.height - 1), -dx, -dy);
+}
+
+// HeadPoseVisualizer
+HeadPoseVisualizer::HeadPoseVisualizer(float scale, cv::Scalar xAxisColor, cv::Scalar yAxisColor, cv::Scalar zAxisColor, int axisThickness):
+                        xAxisColor(xAxisColor), yAxisColor(yAxisColor), zAxisColor(zAxisColor), axisThickness(axisThickness), scale(scale) {
+}
+
+void HeadPoseVisualizer::buildCameraMatrix(cv::Mat& cameraMatrix, int cx, int cy, float focalLength) {
+    if (!cameraMatrix.empty()) return;
+    cameraMatrix = cv::Mat::zeros(3, 3, CV_32F);
+    cameraMatrix.at<float>(0) = focalLength;
+    cameraMatrix.at<float>(2) = static_cast<float>(cx);
+    cameraMatrix.at<float>(4) = focalLength;
+    cameraMatrix.at<float>(5) = static_cast<float>(cy);
+    cameraMatrix.at<float>(8) = 1;
+}
+
+void HeadPoseVisualizer::draw(cv::Mat& frame, cv::Point3f cpoint, double yaw, double pitch, double roll) {
```

Comment:
Why are these parameters `double`s if all calculations are done in single precision?

Reply:
Changed `float` to `double`